mods/auto-theme-switcher.wh.cpp

```diff
@@ -0,0 +1,518 @@
+        return false;
+    if (RegGetValueW(HKEY_CURRENT_USER, L"Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize", L"SystemUsesLightTheme", RRF_RT_REG_DWORD, nullptr, &systemLight, &dataSize) != ERROR_SUCCESS)
+        return false;
+
+    std::wifstream file(themePath);
+    if (!file)
+        return false;
+
+    bool inVisualStyles = false;
+    std::wstring line, systemMode, appMode;
+    while (std::getline(file, line)) {
+        if (line.empty() || line[0] == L';') continue;
+        if (line[0] == L'[') {
+            inVisualStyles = (line == L"[VisualStyles]");
+            continue;
+        }
+        if (inVisualStyles) {
+            if (line.find(L"SystemMode=") == 0)
+                systemMode = line.substr(11);
+            else if (line.find(L"AppMode=") == 0)
+                appMode = line.substr(8);
+            if (!systemMode.empty() && !appMode.empty())
+                break;
+        }
+    }
+
+    auto isLight = [](const std::wstring& s) { return _wcsicmp(s.c_str(), L"Light") == 0; };
+    bool themeLight = isLight(systemMode) && isLight(appMode);
+
+    return (appsLight == (themeLight ? 1 : 0)) && (systemLight == (themeLight ? 1 : 0));
+}
+
+void ApplyAppearance(Appearance appearance) {
+    DWORD val = (appearance == light) ? 1 : 0;
+
+    // change appearance
+    RegSetKeyValueW(HKEY_CURRENT_USER, L"Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize", L"AppsUseLightTheme", REG_DWORD, &val, sizeof(val));
+    RegSetKeyValueW(HKEY_CURRENT_USER, L"Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize", L"SystemUsesLightTheme", REG_DWORD, &val, sizeof(val));
+
+    // broadcast the change
+    SendMessageTimeoutW(HWND_BROADCAST, WM_SETTINGCHANGE, 0, (LPARAM)L"ImmersiveColorSet", SMTO_ABORTIFHUNG, 100, nullptr);
+
+    Wh_Log(L"[Theme] Applied %s Mode.", appearance == light ? L"Light" : L"Dark");
+
+    // apply wallpaper to lock screen
+    if (Wh_GetIntSetting(L"LockScreen"))
+        ApplyLockScreen();
+}
+
+// Based on:
+// https://github.com/qwerty12/AutoHotkeyScripts/blob/a9423f59c945a3a031cb38b25cf461a34de9a6d3/SetThemeFromdotThemeFile.ahk
+void ApplyTheme(PCWSTR themePath) {
+    // {C04B329E-5823-4415-9C93-BA44688947B0}
+    constexpr winrt::guid CLSID_IThemeManager{
+        0xC04B329E,
+        0x5823,
+        0x4415,
+        {0x9C, 0x93, 0xBA, 0x44, 0x68, 0x89, 0x47, 0xB0}};
+
+    // {0646EBBE-C1B7-4045-8FD0-FFD65D3FC792}
+    constexpr winrt::guid IID_IThemeManager{
+        0x0646EBBE,
+        0xC1B7,
+        0x4045,
+        {0x8F, 0xD0, 0xFF, 0xD6, 0x5D, 0x3F, 0xC7, 0x92}};
+
+    winrt::com_ptr<IUnknown> pThemeManager;
+    HRESULT hr =
+        CoCreateInstance(CLSID_IThemeManager, nullptr, CLSCTX_INPROC_SERVER,
+                        IID_IThemeManager, pThemeManager.put_void());
+    if (FAILED(hr) || !pThemeManager) {
+        Wh_Log(L"[Theme] Failed to apply theme.");

```

Comment:
Not returning here will cause a crash. I'd keep `ApplyTheme` as it originally is and call `ApplyLockScreen` outside of it.